pkg/scheduler/api/node_info.go

```diff
@@ -126,6 +128,16 @@ func (ni *NodeInfo) setNodeState(node *v1.Node) {
 		return
 	}
 
+	var nodeInfo cache.NodeInfo
+	nodeInfo.SetNode(ni.Node)
+	if fit, _, _ := predicates.CheckNodeConditionPredicate(nil, nil, &nodeInfo); !fit {
```

Comment:
How to work together with `TaintNodeByCondition`?

Reply:
xref: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/taint-node-by-condition.md 
the example at the end of this document may be help.

Reply:
now  in AddPredicateFn we use func CheckNodeConditionPredicate, if conditions have some type and status, like NodeNetworkUnavailable, true, the pod can't be scheduled.

so i move the CheckNodeConditionPredicate for node forward.

Reply:
@k82cn please help to review :)